core/src/sumeragi/main_loop.rs

```diff
@@ -805,14 +798,19 @@ fn reset_state(
     while *old_view_change_index < current_view_change_index {
         *old_view_change_index += 1;
         error!(addr=%peer_id.address, "Rotating the entire topology.");
-        current_topology.rotate_all();
         was_commit_or_view_change = true;
     }
 
     // Reset state for the next round.
     if was_commit_or_view_change {
         *old_latest_block_height = current_latest_block_height;
 
+        *current_topology = Topology::recreate_topology(
+            &lastest_block,
+            current_view_change_index as usize,
```

Comment:
Please, use failing conversion and `expect()`. We do that in other places and I find it more explicit